### PR TITLE
Use 'jenkins-button' for tables

### PR DESCRIPTION
**Pull request by janfaracik**
Small one to replace `.jenkins-table__button` with just `.jenkins-button`. The two classes are super similar and we're not getting any advantages from having them separated. 

Couple of visual changes:
* Buttons with a colour applied to them now reflect the colour in the buttons background

**Before**
image

**After**
image

* Buttons are slightly bubblier/rounded

### Testing done

* All buttons in core have had their classes changed and look as expected

### Proposed changelog entries

- N/A

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [x] The Jira issue, if it exists, is well-described.
- [x] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [x] There is automated testing or an explanation as to why this change has no tests.
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [x] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [x] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [x] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [x] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

File: core/src/main/resources/hudson/views/BuildButtonColumn/column.jelly
Comment (janfaracik) [inline core/src/main/resources/hudson/views/BuildButtonColumn/column.jelly:26]: This needs adjusting

**Review by timja — APPROVED** [review]
LGTM, I checked the changes here and I looked for any plugins that looked funny and didn't find any but the class hasn't been removed so I wouldn't expect any

**Review by NotMyFault — APPROVED** [review]

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

File: war/src/main/scss/components/_icons.scss
Comment (daniel-beck) [inline war/src/main/scss/components/_icons.scss:130]: Seems unrelated?

File: war/src/main/scss/components/_side-panel-widgets.scss
Comment (janfaracik) [reply war/src/main/scss/components/_side-panel-widgets.scss:249]: The new button is slightly larger and it breaks the build history widget.

(without the fix)
image

(with the fix)
image

File: war/src/main/scss/components/_table.scss
Comment (janfaracik) [reply war/src/main/scss/components/_table.scss:174]: Added a comment in the file - it increases the size of the icons so that they're larger than in regular buttons.

File: war/src/main/scss/components/_icons.scss
Comment (janfaracik) [reply war/src/main/scss/components/_icons.scss:130]: Previously the `svg` was unnecessarily wrapped in a `span`, hence the need to specify `svg`. Seeing as the span has been removed this selector will no longer work - so I've updated it.

File: war/src/main/scss/components/_table.scss
Comment (daniel-beck) [reply war/src/main/scss/components/_table.scss:174]: Right, but this is for non-`.jenkins-button` contents?

Comment (janfaracik) [reply war/src/main/scss/components/_table.scss:174]: Ah sorry I misread - in this case it'll be keeping SVGs the same size between the new buttons and old.

File: core/src/main/resources/hudson/views/BuildButtonColumn/column.jelly
Comment (daniel-beck) [inline core/src/main/resources/hudson/views/BuildButtonColumn/column.jelly:51]: Could you also revert the formatting?

Comment (janfaracik) [reply core/src/main/resources/hudson/views/BuildButtonColumn/column.jelly]: Done

**Review by daniel-beck — APPROVED** [review]

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Comment by mawinter69** [comment]
This causes a regression. The buttons in tables are then no longer resized when the icon size is changed.
The background of the button is also flowing over the row border
Small icons:
image


Large Icons:
image

**Comment by MarkEWaite** [comment]
Thanks for the comment @mawinter69 .  This will first be visible to LTS users in Jenkins 2.462.1 that is scheduled to release 7 Oct 2024.  The impact of the issues does not seem very severe to me, but I agree that it is a change of behavior and it is unexpected that small, medium, and large have very little difference now.

If a fix is provided, should it also be considered for a backport?

If a backport is needed, then an issue needs to be created on https://issues.jenkins.io so that it can be marked as an LTS candidate.

**Comment by mawinter69** [comment]
Created https://issues.jenkins.io/browse/JENKINS-73453

**Pull request merged**
